src/app/sevices/task.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { Task } from '../models/task';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskService{
+  jsonSubscription: Subscription = Subscription.EMPTY;
+  tasksSubscription: Subscription = Subscription.EMPTY;
+  private tasksSubject: BehaviorSubject<Task[]> = new BehaviorSubject<Task[]>([]);
+  tasksObservable$: Observable<Task[]> = this.tasksSubject.asObservable();
```

Comment:
no need to name it tasksObservable$ just tasks$ is fine the $ symbol shows that it's an observable